Index: flexmojos-testing/flexmojos-test-harness/src/test/resources/settings.xml
===================================================================
--- flexmojos-testing/flexmojos-test-harness/src/test/resources/settings.xml	(révision 1866)
+++ flexmojos-testing/flexmojos-test-harness/src/test/resources/settings.xml	(révision 1870)
@@ -26,6 +26,8 @@
           xsi:schemaLocation="http://maven.apache.org/setrings/1.0.0
           http://maven.apache.org/xsd/settings-1.0.0.xsd">

+  <localRepository>${fake-repo}</localRepository>
+
   <profiles>
     <profile>
       <id>flex-extras</id>
Index: flexmojos-testing/flexmojos-test-harness/pom.xml
===================================================================
--- flexmojos-testing/flexmojos-test-harness/pom.xml	(révision 1866)
+++ flexmojos-testing/flexmojos-test-harness/pom.xml	(révision 1870)
@@ -85,21 +85,6 @@
               <outputDirectory>${fake-maven}/lib</outputDirectory>
             </configuration>
           </execution>
-          <execution>
-            <id>copy-repo</id>
-            <phase>generate-test-resources</phase>
-            <goals>
-              <goal>copy-dependencies</goal>
-            </goals>
-            <configuration>
-              <includeGroupIds>org.sonatype.flexmojos</includeGroupIds>
-              <outputDirectory>${fake-repo}</outputDirectory>
-              <useRepositoryLayout>true</useRepositoryLayout>
-              <generateRepositoryMetadatas>true</generateRepositoryMetadatas>
-              <silent>true</silent>
-              <copyPom>true</copyPom>
-            </configuration>
-          </execution>
         </executions>
       </plugin>
       <plugin>
@@ -160,6 +145,7 @@
             <xml>com.adobe.flex.framework:flex-framework:${flex.sdk.version}:pom</xml>

             <xml>org.sonatype.flexmojos:flexmojos-maven-plugin:${project.version}</xml>
+            <xml>org.sonatype.flexmojos:flexmojos-generator-api:${project.version}</xml>
             <xml>org.sonatype.flexmojos:flexmojos-parent:${project.version}:pom</xml>

             <xml>org.sonatype.flexmojos:flexmojos-unittest-support:${project.version}:swc</xml>
Index: flexmojos-maven-plugin/src/main/java/org/sonatype/flexmojos/flexbuilder/FlashBuilderMojo.tmp
===================================================================
--- flexmojos-maven-plugin/src/main/java/org/sonatype/flexmojos/flexbuilder/FlashBuilderMojo.tmp	(révision 1866)
+++ flexmojos-maven-plugin/src/main/java/org/sonatype/flexmojos/flexbuilder/FlashBuilderMojo.tmp	(révision 1870)
@@ -1,91 +0,0 @@
-/**
- *  Copyright 2008 Marvin Herman Froeder
- *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
- *  http://www.apache.org/licenses/LICENSE-2.0
- *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
- *
- */
-package org.sonatype.flexmojos.flexbuilder;
-
-import java.util.Collection;
-
-import org.apache.maven.plugin.MojoExecutionException;
-import org.apache.velocity.VelocityContext;
-
-
-
-
-
-/**
- * Extends the standard Flex Builder configuration.
- *
- * @author Lance Linder (llinder@gmail.com)
- * @since 3.5
- * @extendsPlugin eclipse
- * @extendsGoal eclipse
- * @goal flashbuilder
- * @requiresDependencyResolution
- */
-public class FlashbuilderMojo
-    extends FlexbuilderMojo
-{
-	/**
-	 * IDE Verion. First release of Flashbuilder is version 6.
-	 *
-	 * @parameter default-value=6
-	 */
-	protected Integer ideVersion;
-
-	@Override
-	protected String getAsPropertiesTemplate()
-	{
-		return "/templates/flashbuilder/actionScriptProperties.vm";
-	}
-
-	@Override
-	protected VelocityContext getAsPropertiesContext(ProjectType type, Collection<FbIdeDependency> dependencies)
-		throws MojoExecutionException
-	{
-		VelocityContext context = super.getAsPropertiesContext( type, dependencies );
-
-		context.put( "ideVersion", ideVersion );
-
-		return context;
-	}
-
-	@Override
-	protected String getFlexLibPropertiesTemplate()
-	{
-		return "/templates/flashbuilder/flexLibProperties.vm";
-	}
-
-	@Override
-	protected String getFlexPropertiesTemplate()
-	{
-		return "/templates/flashbuilder/flexProperties.vm";
-	}
-
-	/**
-     * Utility function to sense flex builder SDK value from framework dependencies.
-     *
-     * For example:
-     * 4.0.0 will import into Flexbuilder as "Flex 4.0"
-     * 3.5.0 will import as Flex "3.5".
-     *
-     * This override differs from the base function in that versions such
-     * as 4.0.0 will return 4.0 instead of 4 as in the base implementation.
-     *
-     * @return
-     */
-	@Override
-    protected String getFlexSdkVersion()
-    {
-    	String value = super.getFlexSdkVersion();
-
-    	if( !value.equals( "default" ) && !value.contains(".") )
-    		value += ".0";
-
-    	return value;
-    }
-
-}
Index: flexmojos-maven-plugin/src/main/java/org/sonatype/flexmojos/flexbuilder/FlashbuilderMojo.java
===================================================================
--- flexmojos-maven-plugin/src/main/java/org/sonatype/flexmojos/flexbuilder/FlashbuilderMojo.java	(révision 0)
+++ flexmojos-maven-plugin/src/main/java/org/sonatype/flexmojos/flexbuilder/FlashbuilderMojo.java	(révision 1870)
@@ -0,0 +1,91 @@
+/**
+ *  Copyright 2008 Marvin Herman Froeder
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+package org.sonatype.flexmojos.flexbuilder;
+
+import java.util.Collection;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.velocity.VelocityContext;
+
+
+
+
+
+/**
+ * Extends the standard Flex Builder configuration.
+ *
+ * @author Lance Linder (llinder@gmail.com)
+ * @since 3.5
+ * @extendsPlugin eclipse
+ * @extendsGoal eclipse
+ * @goal flashbuilder
+ * @requiresDependencyResolution
+ */
+public class FlashbuilderMojo
+    extends FlexbuilderMojo
+{
+	/**
+	 * IDE Verion. First release of Flashbuilder is version 6.
+	 *
+	 * @parameter default-value=6
+	 */
+	protected Integer ideVersion;
+
+	@Override
+	protected String getAsPropertiesTemplate()
+	{
+		return "/templates/flashbuilder/actionScriptProperties.vm";
+	}
+
+	@Override
+	protected VelocityContext getAsPropertiesContext(ProjectType type, Collection<FbIdeDependency> dependencies)
+		throws MojoExecutionException
+	{
+		VelocityContext context = super.getAsPropertiesContext( type, dependencies );
+
+		context.put( "ideVersion", ideVersion );
+
+		return context;
+	}
+
+	@Override
+	protected String getFlexLibPropertiesTemplate()
+	{
+		return "/templates/flashbuilder/flexLibProperties.vm";
+	}
+
+	@Override
+	protected String getFlexPropertiesTemplate()
+	{
+		return "/templates/flashbuilder/flexProperties.vm";
+	}
+
+	/**
+     * Utility function to sense flex builder SDK value from framework dependencies.
+     *
+     * For example:
+     * 4.0.0 will import into Flexbuilder as "Flex 4.0"
+     * 3.5.0 will import as Flex "3.5".
+     *
+     * This override differs from the base function in that versions such
+     * as 4.0.0 will return 4.0 instead of 4 as in the base implementation.
+     *
+     * @return
+     */
+	@Override
+    protected String getFlexSdkVersion()
+    {
+    	String value = super.getFlexSdkVersion();
+
+    	if( !value.equals( "default" ) && !value.contains(".") )
+    		value += ".0";
+
+    	return value;
+    }
+
+}
Index: flexmojos-parent/pom.xml
===================================================================
--- flexmojos-parent/pom.xml	(révision 1866)
+++ flexmojos-parent/pom.xml	(révision 1870)
@@ -419,6 +419,11 @@
         <version>1.1.1</version>
       </dependency>
       <dependency>
+        <groupId>org.codehaus.plexus</groupId>
+        <artifactId>plexus-velocity</artifactId>
+        <version>1.1.8</version>
+      </dependency>
+      <dependency>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-eclipse-plugin</artifactId>
         <version>2.5</version>